atomic_reactor/plugins/post_fetch_worker_metadata.py

```diff
@@ -0,0 +1,73 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.plugin import PluginFailedException
+from atomic_reactor.plugins.build_orchestrate_build import get_worker_build_info
+from atomic_reactor.constants import PLUGIN_FETCH_WORKER_METADATA_KEY
+
+
+class FetchWorkerMetadataPlugin(PostBuildPlugin):
+    """
+    Fetch worker metadata from each platform and return a dict of
+    each platform's metadata.
+
+    """
+
+    key = PLUGIN_FETCH_WORKER_METADATA_KEY
+
+    def __init__(self, tasker, workflow):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        """
+        super(FetchWorkerMetadataPlugin, self).__init__(tasker, workflow)
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+
+        metadatas = {}
+
+        # get all the build annotations and labels from the orchestrator
+        build_result = self.workflow.build_result
+
+        if not build_result:
+            raise PluginFailedException('No build_result for FetchWorkerMetadataPlugin')
```

Comment:
Since this is a post-build plugin, it won't run unless the build-step plugin was successful. I think you could either remove this check or replace it with `assert build_result`.